### DISCUSSION TITLE
How to use addon-actions when using it through addon-eesentials

### DISCUSSION THREAD

**Discussion by jim-y** [storybookjs/storybook#26572] [Help]
### Summary

The updated documentation states that using `addon-essentials` gives you `addon-actions` at the essentials page [https://storybook.js.org/docs/essentials](https://storybook.js.org/docs/essentials).

Now i have the essentials enabled in `main.js` as

```
addons: [
    getAbsolutePath('@storybook/addon-webpack5-compiler-swc'),
    getAbsolutePath('@storybook/addon-links'),
    getAbsolutePath('@storybook/addon-essentials')
],
```

yet, when I try to import `action` as

```typescript
import { action } from '@storybook/addon-actions';
```

I get an error `TS2307: Cannot find module @storybook/addon-actions or its corresponding type declarations.` as it is not present in `node_modules`. How should we import the `actions` function then? I tried to replace `action` with `fn` from `@storybook/test` but for my use-case I was not able to do so.

I also installed `addon-actions` separately as `npm i -D @storybook/addon-actions` then adding it to main.js manually, but I got a warning in chrome console that addon-actions was loaded twice. I guess once by addon-essentials then addon-actions

Cheers

**Comment by dsimmons** [comment]
I'm also finding this confusing!

On the one hand, from the documentation, it sounds like I shouldn't need to install this package. On the other, to be able to import `action`, it seems like I ***do*** need to add it manually?

**Comment by warriordog** [comment]
I was able to fix it by following these docs: https://storybook.js.org/docs/essentials/actions#exports
